gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py

```diff
@@ -0,0 +1,109 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Variant merge stategy that can handle both Variants and Non-variants."""
+
+from __future__ import absolute_import
+
+from intervaltree import IntervalTree
+
+from gcp_variant_transforms.beam_io.vcfio import Variant
+from gcp_variant_transforms.libs.variant_merge import move_to_calls_strategy
+from gcp_variant_transforms.libs.variant_merge import variant_merge_strategy
+
+__all__ = ['MergeWithNonVariantsStrategy']
+
+DEFAULT_WINDOW_SIZE = 2500
+
+
+class MergeWithNonVariantsStrategy(variant_merge_strategy.VariantMergeStrategy):
+  """A merging strategy that can handle non-variant records.
+  Variants will be merged across files by using reference_name and start
+  position as key. INFO fields should be moved to class if they match
+  `info_keys_to_move_to_calls_regex`. Otherwise, one will be chosen as
+  representatve (in no particular order) among the merged variants.
+  Filters will be merged across all variants matching the key and the
+  highest quality score will be chosen as representative for the merged
+  variants. The filters and quality fields can be optionally copied to
+  their associated calls using `copy_quality_to_calls` and
+  `copy_filter_to_calls` options.
+  Note: if a field is set to be moved from INFO to calls, then it must
+  not already exist in calls (i.e. specified by FORMAT in the VCF
+  header).
+  """
+
+  def __init__(self, info_keys_to_move_to_calls_regex, copy_quality_to_calls,
+               copy_filter_to_calls, window_size=DEFAULT_WINDOW_SIZE):
+    """Initializes the strategy.
+    Args:
+      info_keys_to_move_to_calls_regex (str): A regular expression
+        specifying info fields that should be moved to calls.
+      copy_quality_to_calls (bool): Whether to copy the quality field to
+        the associated calls in each record.
+      copy_filter_to_calls (bool): Whether to copy filter field to the
+        associated calls in each record.
+      window_size (int): Size of windows that variants will be grouped
+        in based on the start position of the variant
+    """
+    self._move_to_calls_strategy = move_to_calls_strategy.MoveToCallsStrategy(
+        info_keys_to_move_to_calls_regex=info_keys_to_move_to_calls_regex,
+        copy_quality_to_calls=copy_filter_to_calls,
+        copy_filter_to_calls=copy_filter_to_calls)
+    self._window_size = window_size
+
+  def _union(self, list1, list2):
+    return sorted(set(list1 + list2))
+
+  def _merge(self, v1, v2):
+    if v1.alternate_bases == v2.alternate_bases:
```

Comment:
Please check start and end as well (also add a comment that we assume reference_bases match if start and end are the same).

Reply:
Done.